pandas-stubs/core/reshape/concat.pyi

```diff
@@ -12,15 +15,18 @@ from pandas import (
 
 from pandas._typing import HashableT
 
+_HashableT2 = TypeVar("_HashableT2", bound=Hashable)
+_HashableT3 = TypeVar("_HashableT3", bound=Hashable)
+
 @overload
 def concat(
     objs: Iterable[DataFrame] | Mapping[HashableT, DataFrame],
     axis: Literal[0, "index"] = ...,
     join: str = ...,
     ignore_index: bool = ...,
-    keys=...,
-    levels=...,
-    names=...,
+    keys: Sequence[_HashableT2] = ...,
+    levels: list[Sequence] = ...,
```

Comment:
`list[list[HashableT2] | tuple[HashableT2, ...]]` might be better

Reply:
Went with `Sequence[list[HashableT2] | tuple[HashableT2, ...]` due to invariance.